apps/drain-safe/src/utils/gateway.ts

```diff
@@ -0,0 +1,29 @@
+import { fetchJson } from '.';
+
+export interface Asset {
+  balance: string;
+  fiatBalance: string;
+  fiatConversion: string;
+  tokenInfo: {
+    address: string;
+    decimals: number;
+    logoUri: string | null;
+    name: string;
+    symbol: string;
+    type: string;
+  };
+}
+
+interface Balance {
+  fiatTotal: string;
+  items: Asset[];
+}
+
+export const CURRENCY = 'USD';
+
+export async function fetchSafeAssets(safeAddress: string, safeNetwork: string): Promise<Balance> {
+  const network = safeNetwork.toLowerCase() === 'mainnet' ? 'mainnet' : 'rinkeby';
+  const url = `https://safe-client.${network}.gnosis.io/v1/safes/${safeAddress}/balances/${CURRENCY}/?trusted=false&exclude_spam=true`;
```

Comment:
While this is indeed the easiest way, be aware that the client gateway is for the client, so we will deprecate endpoints that are not used by the mobile clients or react interface. We should find a better way for this in the future. I don't think it is blocking for now.

Reply:
I'll check if I can use the transactions API directly.

Reply:
What I meant is that theoretically you can use events to determine what tokens have been used with this address and fetch the balances for these tokens (that would be the version that does not rely on our services at all) :D

Again I don't think this is required for an initial version.